src/undocumented_names.jl

```diff
@@ -0,0 +1,26 @@
+"""
+    test_undocumented_names(module::Module)
+
+Test that all public names in `module` have a docstring (not including the module itself).
+
+!!! warning
+    For Julia versions before 1.11, this does not test anything.
+"""
+function test_undocumented_names(m::Module)
+    @static if VERSION >= v"1.11"
+        undocumented_names = filter(n -> n != nameof(m), Docs.undocumented_names(m))
```

Comment:
```suggestion
        undocumented_names = filter(n -> n != nameof(m), Docs.undocumented_names(m))
```
Why the filter?   It's good practice for a module to have a docstring.

Reply:
context: https://github.com/JuliaTesting/Aqua.jl/pull/313#discussion_r1855426913

Reply:
Ideally that context would be added as a comment here, so that future readers of the code also have a chance to know it :-)